rules/no-missing-di.js

```diff
@@ -0,0 +1,107 @@
+const {
+  is
+} = require('bpmnlint-utils');
+
+const {
+  flatten
+} = require('min-dash');
+
+/**
+ * A rule that checks that there is no BPMNDI information missing for elements,
+ * which require BPMNDI.
+ */
+module.exports = function() {
+
+  function check(node, reporter) {
+
+    if (!is(node, 'bpmn:Definitions')) {
+      return false;
+    }
+
+    // (1) Construct array of all BPMN elements
+    const bpmnElements = getAllBpmnElements(node);
+
+    // (2) Filter BPMN elements without visual representation
+    const visualBpmnElements = bpmnElements.filter(hasVisualRepresentation);
+
+    // (3) Construct array of BPMNDI references
+    const diBpmnReferences = getAllDiBpmnReferences(node);
+
+    // (4) Report elements without BPMNDI
+    visualBpmnElements.forEach((element) => {
+      if (diBpmnReferences.indexOf(element.id) === -1) {
+        reporter.report(element.process.id,
+          `No bpmndi information found for visual representation of <${element.type}> with id=${element.id}`);
+      }
+    });
+  }
+
+  return {
+    check
+  };
+
+};
+
+
+// helpers /////////////////////////////
+
+/**
+ * Get all BPMN elements within a bpmn:Definitions node
+ *
+ * @param {ModdleElement} definitionsNode - A moddleElement representing
+ * the bpmn:Definitions element
+ * @return {array<Object>} A flat array with all BPMN elements part of this
+ * bpmn:Definitions node, each represented with { id: elementId, type: elementType, process: {id: processId} }
+ *
+ */
+function getAllBpmnElements(definitionsNode) {
+  return flatten(
+    definitionsNode.rootElements.map((rootElement) => {
+
+      // Include flowElements (e.g., tasks, sequenceFlows) and participants
+      const elements = rootElement.flowElements || rootElement.participants;
```

Comment:
Maybe I'm missing something here? How do we currently detect the following entities?

* Group, missing DI
* Lanes, missing DI
* Nested sub-processes or their children, missing DI

I would have expected to see some recursion, but I did not find it.

----

Example scenario I'm thinking of:

![image](https://user-images.githubusercontent.com/58601/105381887-ca1b8c80-5c0f-11eb-8d4e-9b261519acb3.png)

[Diagram source](https://github.com/bpmn-io/bpmnlint/files/5850760/diagram.bpmn.txt)

Reply:
Thanks. I was too quick there. Should work now. 

I could not create a case for multiple lanesets :
```
(rootElement.laneSets && rootElement.laneSets[0].lanes) || [])); // is there any case where we can have multiple laneSets?
```

Will check that tomorrow

Reply:
Yes, multiple lane sets can exist, in theory.

Sub-processes can also have lanes.

Reply:
As you are traversing the document, it may also make sense to check this rule against an empty BPMN file. It should not blow up.

Reply:
Added as test